src/libpriv/rpmostree-core.c

```diff
@@ -1022,6 +1029,7 @@ rpmostree_dnf_add_checksum_goal (GChecksum *checksum,
 
   pkglist = hy_goal_list_installs (goal, NULL);
   g_assert (pkglist);
+  g_ptr_array_sort (pkglist, (GCompareFunc)pkg_array_compare);
```

Comment:
But the actual checksum array we use for calculating the final SHA-512 is sorted just down there. So the order in which we add to that array shouldn't matter, no?